### PR TITLE
Goal window formulas tips

### PR DESCRIPTION
**Pull request by barkas90**
New goal window with formulas

### CODE REVIEW

File: Desktop/AddGoalWindow.cs [hunk 1]
```diff
@@ -5,23 +5,86 @@
...
+
+        private void label2_Click(object sender, EventArgs e)
+        {
+
+        }
+
+        private void goalDate_ValueChanged(object sender, EventArgs e)
+        {
+
+        }
```
Comment (AleksandrasMaliuginas) [inline Desktop/AddGoalWindow.cs:83]: Are these functions needed? For what purpose?

File: Desktop/AddTransactionWindow.cs [hunk 2]
```diff
@@ -76,5 +76,20 @@ private void addNewTransactionButton_Click(object sender, EventArgs e)
...
+
+        private void transactionAmount_TextChanged(object sender, EventArgs e)
+        {
+
+        }
+
+        private void transactionDetailsReasons_TextChanged(object sender, EventArgs e)
+        {
+
+        }
```
Comment (AleksandrasMaliuginas) [inline Desktop/AddTransactionWindow.cs:93]: What are these methods made for?

File: Desktop/AddGoalWindow.cs [hunk 3]
```diff
@@ -5,23 +5,86 @@
...
+            if (ch == 46 && goalMoney.Text.IndexOf('.') != -1)
+            {
+                e.Handled = true;
+                return;
+            }
+            if (!Char.IsDigit(ch) && ch != 8 && ch != 46)
+            {
+                e.Handled = true;
+            }
+        }
```
Comment (AleksandrasMaliuginas) [inline Desktop/AddGoalWindow.cs:35]: Maybe it could be done in shorter way?

**Review by imila254 — APPROVED** [review]

File: Desktop/AddTransactionWindow.cs (hunk 2 above)
Comment (barkas90) [reply Desktop/AddTransactionWindow.cs:93]: These are automatically made

File: Desktop/AddGoalWindow.cs [hunk 4]
```diff
@@ -5,23 +5,86 @@
...
+
         private void button1_Click(object sender, EventArgs e)
+        { 
+            DateTime date = goalDate.Value;
+            string amount = goalMoney.Text;
+            string name = goalNameBox.Text;
+
+            ValidateFields(amount, name);
+
+            if (!String.IsNullOrWhiteSpace(amount) && !String.IsNullOrWhiteSpace(name))
```
[the remark this answers is not in the document]
Comment (barkas90) [reply Desktop/AddGoalWindow.cs:57]: Do you have in mind that "IsNullOrWhiteSpace" function needs to be placed in bool for quicker use?

File: Desktop/AddGoalWindow.cs (hunk 1 above)
Comment (barkas90) [reply Desktop/AddGoalWindow.cs:83]: These have no functions have no purpose, I will delete them

File: Desktop/AddGoalWindow.cs (hunk 4 above)
[the remark this answers is not in the document]
Comment (AleksandrasMaliuginas) [reply Desktop/AddGoalWindow.cs:57]: Not sure. But inside `ValidateFields` method you do the same validasion as in if statment here.

File: Desktop/AddTransactionWindow.cs (hunk 2 above)
Comment (AleksandrasMaliuginas) [reply Desktop/AddTransactionWindow.cs:93]: These also can be deleted as the ones written above.

Comment (barkas90) [reply Desktop/AddTransactionWindow.cs:93]: I will delete them

File: Desktop/AddGoalWindow.cs (hunk 3 above)
Comment (barkas90) [reply Desktop/AddGoalWindow.cs:35]: mhm.. I don't think there's a point of short it

**Review by AleksandrasMaliuginas — APPROVED** [review]
Code is **very** similar as in Main.cs. We should think of having an Interface or Abstract class.

**Review by sk057 — APPROVED** [review]

**Pull request merged**